### PR TITLE
Update group page permissions checking & user handler permissions

### PR DESCRIPTION
**Pull request by acrellin**

### CODE REVIEW

File: skyportal/tests/api/test_user.py [hunk 1]
```diff
@@ -10,11 +10,6 @@ def test_get_user_info(manage_users_token, user):
     assert data['data']['id'] == user.id
 
 
-def test_get_user_info_access_denied(view_only_token, user):
```
Comment (acrellin) [inline skyportal/tests/api/test_user.py:13]: Anyone can fetch users now

File: skyportal/handlers/api/group.py [hunk 2]
```diff
@@ -21,6 +21,23 @@
...
+    )
+    return (
+        len(
+            {"System admin", "Manage users", "Manage groups"}.intersection(
+                set(user.permissions)
+            )
+        )
+        > 0
+        or groupuser.admin
+    )
```
Comment (dmitryduev) [inline skyportal/handlers/api/group.py:38]: Nice!

File: skyportal/handlers/api/group.py [hunk 3]
```diff
@@ -454,20 +458,7 @@ def delete(self, group_id, username):
...
-        if (
-            len(
-                {"System admin", "Manage users", "Manage groups"}.intersection(
-                    set(self.current_user.permissions)
-                )
-            )
-            == 0
-            and not current_groupuser.admin
-        ):
+        if not has_admin_access_for_group(self.associated_user_object, group_id):
```
Comment (dmitryduev) [inline skyportal/handlers/api/group.py:461]: So much cleaner!

File: static/js/components/Group.jsx [hunk 4]
```diff
@@ -199,17 +199,20 @@ const Group = () => {
...
-  const isAdmin = (aUser, aGroup) =>
-    aUser &&
-    aGroup.group_users &&
-    aGroup.group_users.filter(
-      (group_user) => group_user.user_id === aUser.id
-    )[0].admin;
+  const isAdmin = (aUser) => {
+    const currentGroupUser = group?.users?.filter(
+      (group_user) => group_user.username === aUser.username
+    )[0];
```
Comment (dmitryduev) [inline static/js/components/Group.jsx:205]: Could there ever be a situation when this is length=0?

File: static/js/components/Group.jsx [hunk 5]
```diff
@@ -310,10 +289,7 @@ const Group = () => {
           <Divider />
           <div className={classes.paper}>
             {/*eslint-disable */}
-            {(currentUser.roles.includes("Super admin") ||
-              isAdmin(currentGroupUser, group)) && (
-              <NewGroupUserForm group_id={group.id} />
-            )}
+            {isAdmin(currentUser) && <NewGroupUserForm group_id={group.id} />}
```
Comment (dmitryduev) [inline static/js/components/Group.jsx:292]: Much cleaner, like!

**Review by dmitryduev — APPROVED** [review]
This is so much cleaner now, thanks a lot @acrellin

**Comment by acrellin** [comment]
GA is green, Travis has unrelated known failures:
```
FAILED skyportal/tests/frontend/test_newsfeed.py::test_news_feed_prefs_widget
FAILED skyportal/tests/tools/test_offset_util.py::test_get_desi_finding_chart
```

File: static/js/components/Group.jsx (hunk 4 above)
Comment (acrellin) [reply static/js/components/Group.jsx:205]: @dmitryduev hmm, I'll have the check this for the case of a non-group-member System admin, will update if necessary. thanks for pointing this out

File: skyportal/handlers/api/group.py [hunk 6]
```diff
@@ -21,6 +21,23 @@
...
 
 
+def has_admin_access_for_group(user, group_id):
+    groupuser = (
+        GroupUser.query.filter(GroupUser.group_id == group_id)
+        .filter(GroupUser.user_id == user.id)
+        .first()
+    )
+    return (
+        len(
```
Comment (stefanv) [inline skyportal/handlers/api/group.py:31]: No need to check length, set will evaluate to True unless it is empty

File: static/js/components/Group.jsx (hunk 4 above)
Comment (acrellin) [reply static/js/components/Group.jsx:205]: @dmitryduev I just checked and this does not error if the list is length 0, it just returns `undefined`, which is perfect for this check

Comment (stefanv) [reply static/js/components/Group.jsx:205]: Can also use `shift()` to get first element or null.

**Review by stefanv — APPROVED** [review]
So nice, Ari!

Left comments, go ahead and merge when you're satisfied.

File: skyportal/handlers/api/group.py (hunk 6 above)
Comment (acrellin) [reply skyportal/handlers/api/group.py]: Good suggestion -- implemented

File: static/js/components/Group.jsx (hunk 4 above)
Comment (stefanv) [reply static/js/components/Group.jsx:205]: Works for me.

File: skyportal/handlers/api/group.py [hunk 7]
```diff
@@ -27,15 +27,9 @@ def has_admin_access_for_group(user, group_id):
...
-    return (
-        len(
-            {"System admin", "Manage users", "Manage groups"}.intersection(
-                set(user.permissions)
-            )
-        )
-        > 0
-        or groupuser.admin
-    )
+    return {"System admin", "Manage users", "Manage groups"}.intersection(
```
Comment (stefanv) [inline skyportal/handlers/api/group.py:30]: Did I lead you "from the road into the ditch" as they say in Afrikaans?  I guess this now needs a bool() to always evalute to True/False :grimacing:

**Comment by acrellin** [comment]
GA is green, Travis has unrelated flaky failures
Merging

**Pull request merged**